staging/src/k8s.io/apimachinery/pkg/runtime/serializer/yaml/meta_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package yaml
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type expectedGVK struct {
+	group   string
+	version string
+	kind    string
+}
+
+var configs = []struct {
+	data string
+	pass bool
+}{ // 0
+	{`apiVersion: v1
+kind: Service
+`, true,
+	},
+	{ // 1
+		`apiVersion: core/v2
+kind: Deployment
+`, true,
+	},
+	{ // 2
+		`apiVersion: v1
+`, true,
+	},
+	{ // 3
+		`kind: Service
+`, true,
+	},
+	{``, true},     // 4
+	{`---`, true},  // 5
+	{`foo`, false}, // 6
+	{ // 7
+		`---
+apiVersion: v1
+kind: Service
+---
+apiVersion: v2
+kind: Deployment
+`, true,
+	},
+}
+
+func helperFailMsg(t *testing.T, msg string, expected, actual interface{}) {
+	t.Errorf("%s -- expected: %#v, actual: %#v", msg, expected, actual)
+}
+
+func helperVerifyValues(t *testing.T, gvk *schema.GroupVersionKind, expected expectedGVK) {
+	if gvk.Group != expected.group {
+		helperFailMsg(t, fmt.Sprintf("GVK mismatch of '%#v'", gvk), "", gvk.Group)
+	}
+	if gvk.Kind != expected.kind {
+		helperFailMsg(t, fmt.Sprintf("GVK mismatch of '%#v'", gvk), "", gvk.Kind)
+	}
+	if gvk.Version != expected.version {
+		helperFailMsg(t, fmt.Sprintf("GVK mismatch of '%#v'", gvk), "", gvk.Version)
+	}
+}
+
+func TestInterpret(t *testing.T) {
```

Comment:
the existing pattern for unit tests is slightly different:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version_test.go#L138-L161

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/runtime/scheme_test.go#L613

- define the test cases as a slice of structs as a inline element of the Test* function.
- all test cases should have names/descriptions - e.g. `name: successfully interpret GVK`.
- iterate over all test cases and call Interpret() tests on them.

Reply:
Ah thanks for pointing me towards that! I adjusted the test cases and I like them much better now.